papers/Sepideh_Alassi_Bertram_Winter/PyCells_for_an_Open_Semiconductor_Industry

```diff
@@ -0,0 +1,270 @@
+	   :scale: 50%
+	   :align: center
+   
+	   Design rules check :label:`DRC` 
+	
+Layout versus Schematic (LVS)
++++++++++++++++++++++++++++++
+The layout of a design should match with its schematic with respect to type, number of devices, connections and topological parameters. In this verification process, we ensure that the layout of a sample design which is generated by PyCells matches perfectly with its corresponding schematic. The LVS tool, first extracts the netlists of layout and schematic of a design according to design rules defined by semiconductor manufacturer. In these netlists, the devices used in design and their connections as well as topological parameters (such as area, perimeter, etc) are listed. The LVS tool compares these netlists and reports if the schematic and layout of the design match completely with respect to type and connection of devices as well as topology of them. It also reports the existence of fragmented nodes which appear if the connections are not identical in the front- and back-end of the design.  
+
+Schematic Driven Layout (SDL)
++++++++++++++++++++++++++++++
+PyCell codes must include the definition of pins regarding type of connections, weak connection (by polysilicon) or strong connection (by metal), and also the definition of shapes requiring external connections. Having types of connections implemented in PyCells, the SDL Navigator tool can attain the schematic of a sample design and generate the layout of it by calling PyCells with respective parameters. After generation of the layout, SDL Navigator checks the connectivity features using flylines.
+
+Database Comparison (DBCOMP)
+++++++++++++++++++++++++++++++
+The last verification step is to substantiate the main aim of this project, generating layouts with Python (PyCells) precisely similar to those generated with SKILL (PCells). This verification process is in GDSII level and is performed with the help of a regression test, by which the PCells and PyCells of identical devices with a same set of parameters are instanced in a layout, and then these instances are compared to be congruent.
+
+
+Conclusion
+------------
+
+Believing in what Richard Stallman has said "Proprietary software is an injustice", we have successfully developed the layout generators for an open semiconductor industry, giving the designers who use ams AG technologies, opportunity to work with any desired EDA-tools, to share their designs easily with others, and to receive the support they deserve. These PyCells are developed for all high and low voltage devices of ams AG technologies in order to enable the IC-designers to have advanced designs without having constraints because of missing parametrized layouts. Furthermore, by using these PyCells we have managed to reduce the computation time which is extremely important in semiconductor industry. The developed PyCells have passed all the verification tests and in practical use by customers.     
+
+
+References
```

Comment:
The tool which we have used to develop the PyCells is called Ciranova which belongs to Synopsis whose products are not open source. This tool is just used by us for development purpose, and the users of our libraries don't need to have or use this development interface in order to execute the codes, PyCells. The PyCell codes can be executable by any EDA-tool which has Python compiler.